lib/sessionContentHandler.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ *  All Rights Reserved
+ *  MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+ * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+ * OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+'use restrict';
+
+const aadutils = require('./aadutils');
+
+/*
+ * the handler for state/nonce
+ * @identifier        - such as 'nonce' or 'state'
+ * @maxAmout          - the max amout of 'identifier' you want to save in the session, default is 200
+ * @maxAge            - when the 'identifier' in session expires in seconds, default is 1800 (seconds)
+ */ 
+function SessionContentHandler(identifier, maxAmount, maxAge) {
+  if (!identifier || identifier === '')
+    throw new Error('identifier is required to use sessionContentHandler');
```

Comment:
'SessionContentHandler: identifier is a required parameter in sessionContentHandler'